dash_cytoscape/CyLeaflet.py

```diff
@@ -212,10 +205,68 @@ def get_cytoscape_max_zoom(self, leaflet_max_zoom):
             {"id": MATCH, "component": "cyleaflet", "sub": "leaf"}, "invalidateSize"
         ),
         Output({"id": MATCH, "component": "cyleaflet", "sub": "leaf"}, "viewport"),
+        Output({"id": MATCH, "component": "cyleaflet", "sub": "cy-extent"}, "data"),
         Input({"id": MATCH, "component": "cyleaflet", "sub": "cy"}, "extent"),
+        Input({"id": MATCH, "component": "cyleaflet", "sub": "cy"}, "maxZoom"),
+        State({"id": MATCH, "component": "cyleaflet", "sub": "cy-extent"}, "data"),
     )
     clientside_callback(
         ClientsideFunction(namespace="cyleaflet", function_name="transformElements"),
-        Output({"id": MATCH, "component": "cyleaflet", "sub": "cy"}, "elements"),
+        Output(
+            {"id": MATCH, "component": "cyleaflet", "sub": "cy"},
+            "elements",
+            allow_duplicate=True,
+        ),
         Input({"id": MATCH, "component": "cyleaflet", "sub": "elements"}, "data"),
+        prevent_initial_call="initial_duplicate",
     )
+
+
+@callback(
+    Output({"id": MATCH, "component": "cyleaflet", "sub": "elements"}, "data"),
+    Input({"id": MATCH, "component": "cyleaflet", "sub": "cy"}, "elements"),
+    prevent_initial_call=True,
+)
+def update_elements_store(elements):
+    def xy_to_lon_lat(x, y):
+        conversion_factor = 20037508.34
+        double_lon = 360
+        max_lon = 180
+        max_lat = 90
+        lon = (x * max_lon) / conversion_factor
+        lat = (
+            math.atan(math.exp((-y * math.pi) / conversion_factor)) * double_lon
+        ) / math.pi - max_lat
+        return [lon, lat]
+
+    new_elements = []
+    if elements:
+        for e in elements:
+            new_element = e
+            if (
+                "position" in e.keys()
+                and "x" in e["position"].keys()
+                and e["position"]["x"]
+                and e["position"]["y"]
+            ):
+                lon_lat = xy_to_lon_lat(e["position"]["x"], e["position"]["y"])
+                new_element["data"]["lon"] = lon_lat[0]
+                new_element["data"]["lat"] = lon_lat[1]
+            new_elements.append(new_element)
+        return new_elements
+    return no_update
+
+
+@callback(
+    Output({"id": MATCH, "component": "cyleaflet", "sub": "cy"}, "maxZoom"),
+    Input({"id": MATCH, "component": "cyleaflet", "sub": "leaf"}, "children"),
+)
+def update_cyto_max_zoom(leaf_children):
+    if isinstance(leaf_children, list) and len(leaf_children) >= 1:
+        tile_layer = list(filter(lambda x: x["type"] == "TileLayer", leaf_children))[0]
+    else:
+        tile_layer = leaf_children
+    leaflet_max_zoom = 18
+    if "maxZoom" in tile_layer["props"].keys():
+        leaflet_max_zoom = tile_layer["props"]["maxZoom"]
```

Comment:
...and if this can't be made a clientside callback, then use the `get_leaflet_max_zoom` function here -- I'm pretty sure it's the same logic. You'll have to move that function outside of the `CyLeaflet` class but that's fine.